parameter types are not renamed when fixing readability-identifier-naming.TypedefCase

Using clang-tidy, I try to rename typedefs, and find that clang-tidy fixes the header and variable declarations, but not parameter types in function signatures.
It is similar to this issue: https://github.com/llvm/llvm-project/issues/47147
I could not find other open issue, so I report it.

Using clang+llvm-12.0.1-x86_64-linux-gnu-ubuntu-, working with c .h and .c files.

### .clang-tidy
```
Checks: "-*,readability-identifier-naming"
CheckOptions:
    - key:      readability-identifier-naming.TypedefCase
      value:    'camelBack'
    - key:      readability-identifier-naming.TypedefSuffix
      value:    '_t'
```

### test.h
```
typedef struct {
    int x;
} ABC;

```
### test.c
```
#include "test.h"
void fun(ABC a) {
    ABC b;
    size s=sizeof(ABC);
    ABC *c  = (ABC *)NULL;
}
```

is rewritten
### test.h
```
typedef struct {
    int x;
} abc_t;

```
### test.c
```
#include "test.h"
void fun(ABC a) {
    abc_t b;
    size s=sizeof(ABC);
    ABC *c  = (ABC *)NULL;
}
```

Type of parameter `a` not renamed.
Type ABC in sizeof is not renamed.
Pointer Type ABC*  is not renamed.
Type ABC in cast is not renamed.

Interesting:

Clang-tidy 14-17:

```
test.hpp:1:7: warning: invalid case style for type alias 'TypeToRename' [readability-identifier-naming]
using TypeToRename = int;
      ^~~~~~~~~~~~
      typeToRename
```

But --fix does not change parameters, does not update .cpp file, and provide invalid fix:
```
typeToRename f(TypeToRename);
```

But with #66835 this looks like this:
```
test.hpp:1:7: warning: invalid case style for type alias 'TypeToRename' [readability-identifier-naming]
    1 | using TypeToRename = int;
      |       ^~~~~~~~~~~~
      |       typeToRename
    2 |
    3 | void f(TypeToRename);
      |        ~~~~~~~~~~~~
      |        typeToRename
test.cpp:3:8: note: FIX-IT applied suggested code changes
    3 | void f(TypeToRename) {
      |        ^
test.hpp:1:7: note: FIX-IT applied suggested code changes
    1 | using TypeToRename = int;
      |       ^
test.hpp:3:8: note: FIX-IT applied suggested code changes
    3 | void f(TypeToRename);
```

So #66835 will fix this issue also.

--- a/clang-tools-extra/clang-tidy/utils/RenamerClangTidyCheck.cpp
+++ b/clang-tools-extra/clang-tidy/utils/RenamerClangTidyCheck.cpp
@@ -256,26 +256,6 @@ public:
       return true;
     }
 
-    // Fix type aliases in value declarations.
-    if (const auto *Value = dyn_cast<ValueDecl>(Decl)) {
-      if (const Type *TypePtr = Value->getType().getTypePtrOrNull()) {
-        if (const auto *Typedef = TypePtr->getAs<TypedefType>())
-          Check->addUsage(Typedef->getDecl(), Value->getSourceRange(), SM);
-      }
-    }
-
-    // Fix type aliases in function declarations.
-    if (const auto *Value = dyn_cast<FunctionDecl>(Decl)) {
-      if (const auto *Typedef =
-              Value->getReturnType().getTypePtr()->getAs<TypedefType>())
-        Check->addUsage(Typedef->getDecl(), Value->getSourceRange(), SM);
-      for (const ParmVarDecl *Param : Value->parameters()) {
-        if (const TypedefType *Typedef =
-                Param->getType().getTypePtr()->getAs<TypedefType>())
-          Check->addUsage(Typedef->getDecl(), Value->getSourceRange(), SM);
-      }
-    }
-
     // Fix overridden methods
     if (const auto *Method = dyn_cast<CXXMethodDecl>(Decl)) {
       if (const CXXMethodDecl *Overridden = getOverrideMethod(Method)) {
@@ -337,6 +317,11 @@ public:
                         DepMemberRef->getMemberNameInfo().getSourceRange(), SM);
     }
 
+    return true;
+  }
+
+  bool VisitTypedefTypeLoc(const TypedefTypeLoc &Loc) {
+    Check->addUsage(Loc.getTypedefNameDecl(), Loc.getSourceRange(), SM);
     return true;
   }
 

--- a/clang-tools-extra/docs/ReleaseNotes.rst
+++ b/clang-tools-extra/docs/ReleaseNotes.rst
@@ -291,7 +291,9 @@ Changes in existing checks
   warnings when a type's forward declaration precedes its definition.
   Additionally, it now provides appropriate warnings for ``struct`` and
   ``union`` in C, while also incorporating support for the
-  ``Leading_upper_snake_case`` naming convention.
+  ``Leading_upper_snake_case`` naming convention. The handling of ``typedef``
+  has been enhanced, particularly within complex types like function pointers
+  and cases where style checks were omitted when functions started with macros.
 
 - Improved :doc:`readability-implicit-bool-conversion
   <clang-tidy/checks/readability/implicit-bool-conversion>` check to take

--- a/clang-tools-extra/test/clang-tidy/checkers/readability/identifier-naming.cpp
+++ b/clang-tools-extra/test/clang-tidy/checkers/readability/identifier-naming.cpp
@@ -729,3 +729,21 @@ struct forward_declared_as_struct;
 class forward_declared_as_struct {
 };
 
+namespace pr55156 {
+
+typedef enum {
+  VALUE0,
+  VALUE1,
+} ValueType;
+// CHECK-MESSAGES: :[[@LINE-1]]:3: warning: invalid case style for typedef 'ValueType' [readability-identifier-naming]
+// CHECK-FIXES: {{^}}} value_type_t;
+
+#define STATIC_MACRO static
+STATIC_MACRO void someFunc(ValueType a_v1, const ValueType& a_v2) {}
+// CHECK-FIXES: {{^}}STATIC_MACRO void someFunc(value_type_t a_v1, const value_type_t& a_v2) {}
+STATIC_MACRO void someFunc(const ValueType** p_a_v1, ValueType (*p_a_v2)()) {}
+// CHECK-FIXES: {{^}}STATIC_MACRO void someFunc(const value_type_t** p_a_v1, value_type_t (*p_a_v2)()) {}
+STATIC_MACRO ValueType someFunc() {}
+// CHECK-FIXES: {{^}}STATIC_MACRO value_type_t someFunc() {}
+#undef STATIC_MACRO
+}

--- a/clang-tools-extra/test/clang-tidy/checkers/readability/identifier-naming.cpp
+++ b/clang-tools-extra/test/clang-tidy/checkers/readability/identifier-naming.cpp
@@ -731,12 +731,18 @@ class forward_declared_as_struct {
 
 namespace pr55156 {
 
+template<typename> struct Wrap;
+
 typedef enum {
   VALUE0,
   VALUE1,
 } ValueType;
 // CHECK-MESSAGES: :[[@LINE-1]]:3: warning: invalid case style for typedef 'ValueType' [readability-identifier-naming]
 // CHECK-FIXES: {{^}}} value_type_t;
+
+typedef ValueType (*MyFunPtr)(const ValueType&, Wrap<ValueType>*);
+// CHECK-MESSAGES: :[[@LINE-1]]:21: warning: invalid case style for typedef 'MyFunPtr' [readability-identifier-naming]
+// CHECK-FIXES: {{^}}typedef value_type_t (*my_fun_ptr_t)(const value_type_t&, Wrap<value_type_t>*);
 
 #define STATIC_MACRO static
 STATIC_MACRO void someFunc(ValueType a_v1, const ValueType& a_v2) {}
@@ -745,5 +751,7 @@ STATIC_MACRO void someFunc(const ValueType** p_a_v1, ValueType (*p_a_v2)()) {}
 // CHECK-FIXES: {{^}}STATIC_MACRO void someFunc(const value_type_t** p_a_v1, value_type_t (*p_a_v2)()) {}
 STATIC_MACRO ValueType someFunc() {}
 // CHECK-FIXES: {{^}}STATIC_MACRO value_type_t someFunc() {}
+STATIC_MACRO void someFunc(MyFunPtr, const MyFunPtr****) {}
+// CHECK-FIXES: {{^}}STATIC_MACRO void someFunc(my_fun_ptr_t, const my_fun_ptr_t****) {}
 #undef STATIC_MACRO
 }

